### DISCUSSION TITLE
solving some trouble getting setmeanneggain.bat to work on windows 10

### DISCUSSION THREAD

**Discussion by TLmv** [ImageMagick/ImageMagick#6425] [Digital Image Processing] [answered]
i'm running Imagemagick 7.1.1 Q16 HDRI on a windows 10 machine and was trying to change the mean of a file, thanks to a helpfull script from snibgo i tried:

``setmeanneggain.bat IMG_0451.JPG 0.17879 0.0197655``

however, i got no errors back.
but as far as i can tell the file is unchanged, i also see no date modified in windows.
and i'm unsure what i'm missing or doing wrong.

> J:\>rem From image IMG_0451.JPG, creates output 0.17879 setting the mean by the "negative gain" method
new_mn_R=0.0197655 new_mn_G=0.0197655 new_mn_B=0.0197655
> J:\>endlocal   & set smngOUTFILE=0.17879

**Comment by snibgo** [comment, marked as answer]
You have edited your question. To address your original post: My scripts call other scripts. You can download them all at once, from the page [Zipped BAT files](http://im.snibgo.com/zipbats.htm).

`echoOffSave.bat` and `echoRestore.bat` just mess with `echo on` and `echo off`, so you can safely delete calls to those.

The call to...
```
call %PICTBAT%setInOut %1 smng
```
... sets the variable INFILE to %1, and does some other things. So you could change that to:
```
set INFILE=%1
```

The comments at the top show the script parameters you need: the input file, then the output file, then one or three numbers.

**Comment by TLmv** [comment]
yes, i'm sorry i edited my question, i have the full package now.
and got it working.

**Comment by snibgo** [comment]
Good stuff.
